### ISSUE TITLE
Lint/AmbiguousBlockAssociation cites unambiguous lambda param

### ISSUE THREAD

**Issue by cainlevy** [bbatsov/rubocop#4189]
When passing a lambda declaration as a method parameter, the new Lint/AmbiguousBlockAssociation cop forces unnecessary parentheses.

Example:

```ruby
# cited
foo ->(a) { a }

# making the cop happy
foo(->(a) { a })
```

Note that the method does not generally require parentheses:

```ruby
# good
foo a
```

## Expected behavior

I expect Lint/AmbiguousBlockAssociation to not cite lambdas passed as params like any other object.

## Actual behavior

The new Lint/AmbiguousBlockAssociation cop creates a citation for lambdas passed as params like any other object.

## Steps to reproduce the problem

Create file:

```ruby
foo ->(a) { a }
```

Run rubocop:

```shell
rubocop -D
```

## RuboCop version

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.3 x86_64-darwin16)
```

**Comment by smakagon** [comment]
@Drenmi thanks for fixing that! I missed that case...

**Comment by ivanovaleksey** [comment]
I got plain Rails scope like `scope :technical, -> { where(technical: true) }`
and I have 
> Parenthesize the param -> { where(technical: true) } to make sure that the block will be associated with the -> method call.

`scope(:technical, -> { where(technical: true) })` fixes the problem but is it the way it should be?

**Comment by bbatsov** [comment]
@ivanovaleksey Are you using 0.48.1?

**Comment by ivanovaleksey** [comment]
@bbatsov, yes
```
$ bundle exec rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.3.3 x86_64-darwin16)
```

**Comment by brandonweiss** [comment]
Yeah, I'm seeing the same thing, on 0.48.1 for something like:

```ruby
scope :vat, -> { where(post_subtotal_type: 'vat') }
```

**Comment by smakagon** [comment]
@brandonweiss @bbatsov yes, there is a problem with lambda check. It's easily fixable. Should I create another PR to this issue?
